### PR TITLE
Better 404 messages from get_document_or_404() in debug mode

### PR DESCRIPTION
**Pull request by NoahCarnahan**
404 pages triggered by `get_document_or_404()` will now display a short message about the reason for the exception when in debug mode. Can be useful if you don't know where the exception was raised.

screen shot 2015-07-09 at 10 46 09 am

FYI @esoergel.
cc @snopoke

### CODE REVIEW

File: corehq/util/couch.py [hunk 1]
```diff
@@ -18,18 +19,35 @@ def get_document_or_404(cls, domain, doc_id, additional_doc_types=None):
     allowed_doc_types = (additional_doc_types or []) + [cls.__name__]
     try:
         unwrapped = cls.get_db().get(doc_id)
-    except ResourceNotFound:
-        raise Http404()
+    except ResourceNotFound as e:
```
Comment (dimagimon) [inline corehq/util/couch.py]: F841 local variable 'e' is assigned to but never used

**Comment by NoahCarnahan** [comment]
Added traceback to the message. Unfortunately the formatting doesn't come through, so it's a bit hard to read, but might still be useful I think.

File: corehq/util/couch.py (hunk 1 above)
Comment (esoergel) [inline corehq/util/couch.py]: Did you mean to pass this to something?  Looks like `traceback.format_exc()` uses this value anyways, is that right?

**Comment by esoergel** [comment]
Is there a way to do this generically?  Like make a subclass of HTTP404 which includes the trace, or figuring out how to modify the middleware which intercepts that exception.

**Comment by czue** [comment]
(build fail isn't real)

no blockers from me but i'll let @esoergel merge when he's good.

**Comment by esoergel** [comment]
@NoahCarnahan ping.  Merging 'cause this is strictly an improvement, but still had some questions.

**Comment by NoahCarnahan** [comment]
Thanks @esoergel, I'll look into the options for making it more generic.

**Comment by dannyroberts** [comment]
@NoahCarnahan This would have been super useful for http://manage.dimagi.com/default.asp?175848!

**Pull request merged**